functests/sctp/sctp.go

```diff
@@ -480,3 +499,61 @@ func setupEgress(namespace, fromPod, toPod string, port int32) error {
 	_, err := client.Client.NetworkPolicies(namespace).Create(policy)
 	return err
 }
+
+func findSCTPNodeSelector() (string, error) {
+	mcList, err := client.Client.MachineConfigs().List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mc := range mcList.Items {
+		if enablesSCTP(mc) {
+			mcLabel, found := mc.ObjectMeta.Labels["machineconfiguration.openshift.io/role"]
+			if !found {
+				continue
+			}
+
+			newSCTPNodeSelector, err := findSCTPNodeSelectorByMCLabel(mcLabel)
```

Comment:
This means that you exit from the discovery if you find a mc that does not have the selector (while there could be another one down the list that fits our needs).